src/zenml/cli/stack.py

```diff
@@ -174,6 +174,299 @@ def register_stack(
+
+        if secrets_manager_name:
+            stack_components[
+                StackComponentType.SECRETS_MANAGER
+            ] = repo.get_stack_component(
+                StackComponentType.SECRETS_MANAGER,
+                name=secrets_manager_name,
+            )
+
+        if feature_store_name:
+            stack_components[
+                StackComponentType.FEATURE_STORE
+            ] = repo.get_stack_component(
+                StackComponentType.FEATURE_STORE,
+                name=feature_store_name,
+            )
+
+        if model_deployer_name:
+            stack_components[
+                StackComponentType.MODEL_DEPLOYER
+            ] = repo.get_stack_component(
+                StackComponentType.MODEL_DEPLOYER,
+                name=model_deployer_name,
+            )
+
+        stack_ = Stack.from_components(
+            name=stack_name, components=stack_components
+        )
+        repo.update_stack(stack_name, stack_)
+        cli_utils.declare(f"Stack `{stack_name}` successfully updated!")
+
+
+@stack.command(
+    "remove-component", context_settings=dict(ignore_unknown_options=True)
+)
+@click.argument("stack_name", type=str, required=True)
+@click.option(
+    "-c",
+    "--container_registry",
+    "container_registry_flag",
+    help="Include this to remove the container registry from this stack.",
+    is_flag=True,
+    required=False,
+)
+@click.option(
+    "-s",
+    "--step_operator",
+    "step_operator_flag",
+    help="Include this to remove the step operator from this stack.",
+    is_flag=True,
+    required=False,
+)
+@click.option(
+    "-x",
+    "--secrets_manager",
+    "secrets_manager_flag",
+    help="Include this to remove the secrets manager from this stack.",
+    is_flag=True,
+    required=False,
+)
+@click.option(
+    "-f",
+    "--feature_store",
+    "feature_store_flag",
+    help="Include this to remove the feature store from this stack.",
+    is_flag=True,
+    required=False,
+)
+@click.option(
+    "-d",
+    "--model_deployer",
+    "model_deployer_flag",
+    help="Include this to remove the model deployer from this stack.",
+    is_flag=True,
+    required=False,
+)
+def remove_stack_component(
+    stack_name: str,
+    container_registry_flag: Optional[bool] = False,
+    step_operator_flag: Optional[bool] = False,
+    secrets_manager_flag: Optional[bool] = False,
+    feature_store_flag: Optional[bool] = False,
+    model_deployer_flag: Optional[bool] = False,
+) -> None:
+    """Remove stack components from a stack."""
+    with console.status(f"Updating stack `{stack_name}`...\n"):
+        repo = Repository()
+        try:
+            current_stack = repo.get_stack(stack_name)
+        except KeyError:
+            cli_utils.error(
+                f"Stack `{stack_name}` cannot be updated as it does not exist.",
+            )
+        stack_components = current_stack.components
+
+        if container_registry_flag:
+            stack_components = {
```

Comment:
I think this could also be solved by a `stack_components.pop(StackComponentType.CONTAINER_REGISTRY, None)` as well to make it a little shorter. That will also remove a key-value pair from a dictionary in case a value for that key exists.